assets/js/product-category-block.js

```diff
@@ -161,9 +163,50 @@ class ProductByCategoryBlock extends Component {
 		);
 	}
 
+	renderEditMode() {
+		const { setAttributes } = this.props;
+		const { categories } = this.props.attributes;
+		const { categoriesList } = this.state;
+
+		return (
+			<Placeholder
+				icon="category"
+				label={ __( 'Products by Category', 'woocommerce' ) }
+			>
+				{ __(
+					'Display a grid of products from your selected categories',
+					'woocommerce'
+				) }
+				{ categoriesList.length ? (
+					<div className="wc-block-products-category__selection">
+						<TreeSelect
+							label={ __( 'Product Category', 'woocommerce' ) }
+							tree={ categoriesList }
+							selectedId={ categories }
+							multiple
+							onChange={ ( value ) => {
```

Comment:
Yes, calypso has a rule agains this, but Gutenberg seems to use this pattern, and recommends it [in documentation](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/tree-select). I've never investigated why the calypso linter doesn't like this, but i'm happy to change if there's a (performance?) reason for it.

Reply:
I believe the notion in Calypso is it might be making a new function on re-renders ( i have no links to back this up, just faded old memories ), but if core has this pattern then it seems like a moot point to me - as that is the standards i believe we should follow

Reply:
Looks like that was the reason, but [it wasn't a big deal and calypso has removed that rule.](https://github.com/Automattic/eslint-config-wpcalypso/pull/18) Thanks for pointing it out though, good to check on 🙂